Tasks/AzureFileCopy/Utility.ps1

```diff
@@ -255,14 +255,48 @@ function Upload-FilesToAzureContainer
             $blobStorageURI = $blobStorageEndpoint+$containerName+"/"+$blobPrefix
         }
 
+        $azcopy = "AzCopy\AzCopy.exe"
+        Write-Verbose "Calling AzCopy = $azcopy" -Verbose
+
+        $sourceLocationDir = [System.IO.Path]::GetDirectoryName($sourcePath)
+        $filenm = Split-Path $sourcePath -leaf
+        $isFile = Test-Path -Path $sourcePath -PathType Leaf
+
+        $azlog = ("{0}\..\azlog" -f $sourceLocationDir)
+
+        if($isFile){
+            $args = ("/Source:`"{0}`" /Dest:{1} /DestKey:{2} /Pattern:{3}" -f $sourceLocationDir, $blobStorageURI, $storageKey, $filenm)
+        }
+        else{
+            $args = ("/Source:`"{0}`" /Dest:{1} /DestKey:{2}" -f $sourcePath, $blobStorageURI, $storageKey)
+        }
+
         if([string]::IsNullOrWhiteSpace($additionalArguments))
         {
-            $uploadResponse = Copy-FilesToAzureBlob -SourcePathLocation $sourcePath -StorageAccountName $storageAccountName -ContainerName $containerName -BlobPrefix $blobPrefix -StorageAccountKey  $storageKey -AzCopyLocation $azCopyLocation -BlobStorageURI $blobStorageURI
+            $args += (" /S /Z:`"{0}`"" -f $azlog)
```

Comment:
I see a common pattern in If and else block so that could be moved out of if and else.